src/components/slide-toggle/slide-toggle.ts

```diff
@@ -214,6 +240,54 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
     this._change.emit(event);
   }
 
+
+  private _onDragStart() {
+    if (this._dragPointer) {
+      return;
+    }
+
+    let thumbBarEl = this._elementRef.nativeElement.querySelector('.md-slide-toggle-bar');

```

Comment:
Could you move the logic that does the direct DOM manipulation into a separate class called `SlideToggleRenderer`? The constructor should just take the `ElementRef` for the component. The component class can then just call methods on the `slideToggleRenderer` to do the stuff it needs. 

Once we support web-workers with Angular, the main component code and the stuff that touches the DOM will have to be split into separate classes with marshalling between them. Breaking up the code now makes us more ready for that later.